src/interactor/state_machine/interactor_steps/DeleteLabelStep.cc

```diff
@@ -0,0 +1,98 @@
+#include "DeleteLabelStep.h"
+
+#include "interactor/io_facility/Strings.h"
+#include "interactor/state_machine/interactor_steps/PromptStep.h"
+#include "utils/TaskIdUtils.h"
+
+namespace task_manager {
+
+std::unique_ptr<Command> DeleteLabelStep::execute(Context ctx) {
+  --stage_;
+  if (stage_ == 1) {
+    return HandleStage<1>(ctx);
+  }
+  if (stage_ == 0) {
+    return HandleStage<0>(ctx);
+  }
+  std::terminate();
+}
+
+template <>
+std::unique_ptr<Command> DeleteLabelStep::HandleStage<1>(Context &) {
+  if (arg_.empty()) {
+    return ReportError(Strings::kRequiredId);
+  }
+
+  auto token = validator_->ConsumeOneTokenFrom(arg_);
+  auto to_edit = validator_->ParseInt(token);
+  if (!to_edit) {
+    return ReportError(Strings::InvalidId(token));
+  }
+
+  task_id_.set_id(*to_edit);
+  if (arg_.empty()) {
+    return ReportError(Strings::kRequiredLabel);
+  }
+
+  token = validator_->ConsumeOneTokenFrom(arg_);
+  if (!arg_.empty()) {
+    return ReportError(Strings::kMultipleArgumentDoesNotSupported);
+  }
+
+  label_.set_name(token);
+  return std::make_unique<GetSpecifiedTasksCommand>(
+      std::vector<TaskId>{task_id_});
+}
+
+template <>
+std::unique_ptr<Command> DeleteLabelStep::HandleStage<0>(Context &ctx) {
+  if (!ctx.solid_tasks &&
+      ctx.status == ModelController::Status::kNotPresentId) {
+    return ReportError(Strings::NotPresentId(std::to_string(task_id_.id())));
+  }
+
+  auto found =
+      std::find_if(ctx.solid_tasks->cbegin(), ctx.solid_tasks->cend(),
+                   [this](const auto &i) { return i.task_id() == task_id_; });
+
```

Comment:
above is code repetition with respective part of AddLabelStep, can be made a function (also it's the only place ctx is used)